### PR TITLE
week 2 Assignment by Nived

### PR DESCRIPTION
**Pull request by nivedphil**

### CODE REVIEW

File: week_2/pet.rb [hunk 1]
```diff
@@ -88,9 +94,17 @@ def food_cost(days)
...
   # fish = Pet.new(name: 'clownfish', animal_type_id: 2, food_consumed_per_day: 0.1)
   # snake = Pet.new(name: 'python', animal_type_id: 4, food_consumed_per_day: 0.3)
-  # Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
+   #Pet.cost_to_feed([cat, dog, fish, snake], 6) will return 6180.0
+  
   def self.cost_to_feed(pets, days)
-    raise NotImplementedError # TODO
+    def do_the_deed
+      return FOOD_COST_PER_KG[@animal_type_id] * @food_consumed_per_day
+    end
```
Comment (hrushikeshj) [inline week_2/pet.rb:102]: You can reuse the `food_cost` method.

File: week_2/pet.rb [hunk 2]
```diff
@@ -46,13 +46,19 @@ def initialize(name: , animal_type_id: , food_consumed_per_day:)
...
   # Return the habitat of the pet
   def habitat
-    raise NotImplementedError # TODO
+    if @animal_type_id in 1..3
+      return "Aquatic"
+    elsif @animal_type_id == 4
+      return "Terrariums"
+    elsif @animal_type_id in 5..6
+      return "Cages"
+    end
```
Comment (hrushikeshj) [inline week_2/pet.rb:55]: If a new HABITAT is to be added, here, both the `HABITATS` hash and this function has to be updated. Which is not convenient. So, make use of the `HABITATS` hash to find the habitat here.

File: week_2/pet.rb [hunk 3]
```diff
@@ -107,9 +121,21 @@ def self.cost_to_feed(pets, days)
...
+    for animal in pets
+      animal.mapping(result)
+    end
+    def print_result(hash2)
+      hash2[@animal_type_id].push(name)
+    end
+    for pet in pets
+      pet.print_result(result)
+    end
+    return result
```
Comment (hrushikeshj) [inline week_2/pet.rb:139]: `mapping` and `print_result` are not helping you simplify the problem.

File: week_2/pet.rb [hunk 4]
```diff
@@ -107,9 +121,21 @@ def self.cost_to_feed(pets, days)
...
   # Note - Order is not important
   def self.group_by_animal_type(pets)
-    raise NotImplementedError # TODO
+    result = {}
+    def mapping(hash)
+      hash[@animal_type_id] = []
+    end
+    for animal in pets
+      animal.mapping(result)
+    end
```
Comment (hrushikeshj) [inline week_2/pet.rb:132]:
```suggestion
    for animal in pets
      result[animal.animal_type_id] = []
    end
```
You can also avoid using 2 loops by checking if a key already exist
```ruby
if result[animal.animal_type_id]
   #....
else
  result[animal.animal_type_id] = [animal.name]
end
```

**Review by hrushikeshj — COMMENTED** [review]
Good job, I have left some suggestions

File: week_2/pet.rb (hunk 2 above)
Comment (nivedphil) [reply week_2/pet.rb:55]: So.. can you please tell me how to do that?..

**Pull request closed without merging**